### PR TITLE
More changes to the game engine

### PR DESCRIPTION
**Pull request by greyturn**
Randomly generate player strategies OR
Read them from file

Flag to set number of players

Bug fixes

### CODE REVIEW

File: board.py [hunk 1]
```diff
@@ -54,10 +54,12 @@ def valid_vertex_for_position(self, vertex_id):
...
         """
-        #TODO: Check two edges away
-        #for x in self.vertices:
-        #    if (x.owner != None):
-        #        return False
+        this_vertex = self.vertices[vertex_id]
+        
+        for neighbour in this_vertex.neighbours:
+            neighbour_vertex = self.vertices[neighbour-1]
+            if (neighbour_vertex.owner != None):
```
Comment (Hrodgeir) [inline board.py]: Don't need the brackets

File: game_engine.py [hunk 2]
```diff
@@ -121,15 +129,15 @@ def determine_score(tile_type, weight_array):
         :param weight_array: An array of weights that correspond to each tile type and its related strategy
         :return: 1 * the weight
```
Comment (Hrodgeir) [inline game_engine.py:130]: What's the point of multiplying by 1?

File: player.py [hunk 3]
```diff
@@ -7,6 +13,67 @@ def __init__(self, Id, strategy):
...
+    @staticmethod
+    def valid_strategy(strategy):
+        """
+        Ensure the strategy is valid
+        :param strategy: String, The strategy to validate
+        """
+        strategy_catalogue = ["cities", "settlements", "sheep_monopoly", "wheat_monopoly", "stone_monopoly", "brick_monopoly", "wood_monopoly", "development"]
+        
+        for check_strategy in strategy_catalogue:
+            if strategy == check_strategy:
```
Comment (Hrodgeir) [inline player.py]: Can just do a ternary operator:
`return true if strategy in strategy_catalogue else return false`

**Review by Hrodgeir — CHANGES_REQUESTED** [review]

File: game_engine.py (hunk 2 above)
Comment (greyturn) [reply game_engine.py:130]: currently 1 as a placeholder to represent that value being weighted

**Review by gordillo99 — APPROVED** [review]
it's a good first step. still need to account for dice values and docks and maybe other things?

**Review by Hrodgeir — APPROVED** [review]

**Pull request merged**